core/dbt/adapters/sql/impl.py

```diff
@@ -114,6 +115,67 @@ def expand_column_types(self, goal, current):
 
                 self.alter_column_type(current, column_name, new_type)
 
+    def has_schema_changed(self, temp_relation, target_relation):
+        """
+        Look for schema changes between the target columns and reference columns
+        Step through each column and return that a schema change *has* happened
+        if any of the following are true:
+        1. Number of columns are different (column has been added or removed)
+        2. Columns have different names
+        3. Columns have different data type
+        4. Columns have different data type size
+        """
+        reference_columns = {
+            c.name: c for c in
+            self.get_columns_in_relation(relation = temp_relation)
+        }
+
+        target_columns = {
+            c.name: c for c in
+            self.get_columns_in_relation(relation = target_relation)
+        }
+
+        # 1. The schema has changed if columns have been added or removed
+        if len(reference_columns) != len(target_columns):
+            logger.debug("Schema difference detected: column count in reference does not match target. reference columns: {}  target columns: {} ".format(reference_columns, target_columns))
+            logger.debug("target_ref = {} ".format(target_columns))
+            logger.debug("temp_ref = {} ".format(reference_columns))
+            return True
+
+        for reference_column_name, reference_column in reference_columns.items():
+            target_column = target_columns.get(reference_column_name)
+            # 2a. The schema has changed if a reference column is not found in the target columns
+            if target_column is None:
+                logger.debug("Schema difference detected: reference column {} not found in target".format(reference_column_name))
+                logger.debug("target_ref = {}".format(target_columns))
+                logger.debug("temp_ref = {}".format(reference_columns))
+                return True
+
+            # 3/4. If the columns do not have the same data type and size (see core/dbt/schema.py for more details) 
+            refernce_type = re.findall(r"^(\w*)",reference_column.data_type) # just get the name of datatype, not size/precision
```

Comment:
So this was just to support the on-schema-change. What I wound up running into was if the type was the same but the precision changed, it would return a false positive when It would run the test. I think there is already logic in DBT to handle expansion of columns and take the larger of the two when doing an incremental merge, would that no longer work?